libs/testserver/server.go

```diff
@@ -52,6 +69,22 @@ func (s *Server) Handle(pattern string, handler HandlerFunc) {
 			return
 		}
 
+		if s.RecordRequests {
+			body, err := io.ReadAll(r.Body)
+			assert.NoError(s.t, err)
+
+			var reqBody map[string]any
+			err = json.Unmarshal(body, &reqBody)
+			assert.NoError(s.t, err)
+
+			s.requests = append(s.requests, Request{
+				Method: r.Method,
+				Path:   r.URL.Path,
+				Body:   reqBody,
```

Comment:
if unmarshal fails, this would be empty. Should log the original response

Reply:
Done, we send the original response now.

Reply:
Note: This still assumes the request body is JSON. APIs which send binary responses for example would need to be accommodated later.